core/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/core/model/HttpInput.kt

```diff
@@ -0,0 +1,153 @@
+import org.elasticsearch.common.Strings
+import org.elasticsearch.common.xcontent.NamedXContentRegistry
+import org.elasticsearch.common.xcontent.ToXContent
+import org.elasticsearch.common.xcontent.XContentBuilder
+import org.elasticsearch.common.xcontent.XContentParser
+import org.elasticsearch.common.xcontent.XContentParserUtils
+import java.io.IOException
+
+/**
+ * This is a data class of HTTP type of input for Monitors.
+ */
+data class HttpInput(
+    val scheme: String,
+    val host: String,
+    val port: Int,
+    val path: String,
+    val params: Map<String, String>,
+    val url: String,
+    val connection_timeout: Int,
+    val socket_timeout: Int
+) : Input {
+
+    // Verify parameters are valid during creation
+    init {
+        require(areFieldsValid()) {
+            "Either one of url or scheme + host + port + path + params can be set."
+        }
+        require(connection_timeout in 1..5) {
+            "Connection timeout: $connection_timeout is not in the range of 1 - 5"
+        }
+        require(socket_timeout in 1..60) {
+            "Socket timeout: $socket_timeout is not in the range of 1 - 60"
+        }
+
+        // Create an UrlValidator that only accepts "http" and "https" as valid scheme and allows local URLs.
+        val urlValidator = UrlValidator(arrayOf("http", "https"), UrlValidator.ALLOW_LOCAL_URLS)
+
+        // Build url field by field if not provided as whole.
+        val constructedUrl = if (Strings.isEmpty(url)) {
+            val uriBuilder = URIBuilder()
+            uriBuilder.scheme = scheme
+            uriBuilder.host = host
+            uriBuilder.port = port
+            uriBuilder.path = path
+            for (e in params.entries)
+                uriBuilder.addParameter(e.key, e.value)
+            uriBuilder.build()
+        } else {
+            URIBuilder(url).build()
+        }
+
+        require(urlValidator.isValid(constructedUrl.toString())) {
+            "Invalid url: $constructedUrl"
+        }
+    }
+
+    override fun toXContent(builder: XContentBuilder, params: ToXContent.Params): XContentBuilder {
+        return builder.startObject()
+                .startObject(HTTP_FIELD)
+                .field(SCHEME_FIELD, scheme)
+                .field(HOST_FIELD, host)
+                .field(PORT_FIELD, port)
+                .field(PATH_FIELD, path)
+                .field(PARAMS_FIELD, this.params)
+                .field(URL_FIELD, url)
+                .field(CONNECTION_TIMEOUT_FIELD, connection_timeout)
+                .field(SOCKET_TIMEOUT_FIELD, socket_timeout)
+                .endObject()
+                .endObject()
+    }
+
+    override fun name(): String {
+        return HTTP_FIELD
+    }
+
+    companion object {
+        const val SCHEME_FIELD = "scheme"
+        const val HOST_FIELD = "host"
+        const val PORT_FIELD = "port"
+        const val PATH_FIELD = "path"
+        const val PARAMS_FIELD = "query_params"
```

Comment:
I changed the name of PARAMS_FIELD from "params" to "query_params". (Here is the [code](https://github.com/ann3431/alerting/blob/430d718f96803ceec00b5fb50e664300fde6fa21/core/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/core/model/HttpInput.kt#L102) in the original PR)